homeassistant/components/sia/binary_sensor.py

```diff
@@ -101,63 +128,50 @@ def handle_last_state(self, last_state: State | None) -> None:
                 self._attr_available = False
 
 
-class SIABinarySensorMoisture(SIABinarySensorBase):
-    """Class for Moisture Binary Sensors."""
-
-    def __init__(
-        self,
-        entry: ConfigEntry,
-        account_data: dict[str, Any],
-        zone: int,
-    ) -> None:
-        """Initialize a Moisture binary sensor."""
-        super().__init__(entry, account_data, zone, DEVICE_CLASS_MOISTURE)
-        self._attr_entity_registry_enabled_default = False
-
-    def update_state(self, sia_event: SIAEvent) -> None:
-        """Update the state of the binary sensor."""
-        new_state = MOISTURE_CODE_CONSEQUENCES.get(sia_event.code, None)
-        if new_state is not None:
-            _LOGGER.debug("New state will be %s", new_state)
-            self._attr_is_on = new_state
-
-
-class SIABinarySensorSmoke(SIABinarySensorBase):
-    """Class for Smoke Binary Sensors."""
+class SIABinarySensorRegular(SIABinarySensorBase):
+    """Class for Smoke, Moisture, and Power Binary Sensors."""
 
     def __init__(
         self,
         entry: ConfigEntry,
         account_data: dict[str, Any],
         zone: int,
+        enabled: bool,
+        device_class: str,
     ) -> None:
-        """Initialize a Smoke binary sensor."""
-        super().__init__(entry, account_data, zone, DEVICE_CLASS_SMOKE)
-        self._attr_entity_registry_enabled_default = False
+        """Initialize a binary sensor."""
+        super().__init__(entry, account_data, zone, device_class, enabled)
+        self.consequences = CODE_CONSEQUENCES[device_class]
 
     def update_state(self, sia_event: SIAEvent) -> None:
         """Update the state of the binary sensor."""
-        new_state = SMOKE_CODE_CONSEQUENCES.get(sia_event.code, None)
+        new_state = self.consequences.get(sia_event.code, None)
         if new_state is not None:
             _LOGGER.debug("New state will be %s", new_state)
             self._attr_is_on = new_state
 
 
-class SIABinarySensorPower(SIABinarySensorBase):
-    """Class for Power Binary Sensors."""
+class SIABinarySensorConnectivity(SIABinarySensorBase):
+    """Class for Connectivity Binary Sensors."""
 
     def __init__(
         self,
         entry: ConfigEntry,
         account_data: dict[str, Any],
     ) -> None:
-        """Initialize a Power binary sensor."""
-        super().__init__(entry, account_data, SIA_HUB_ZONE, DEVICE_CLASS_POWER)
-        self._attr_entity_registry_enabled_default = True
+        """Initialize a Connectivity binary sensor."""
+        super().__init__(entry, account_data, None, DEVICE_CLASS_CONNECTIVITY, True)
```

Comment:
The device class has been set, in that case, an icon should not be set.

Reply:
Why is this disabled by default? Wouldn't it make more sense to mark it as the diagnostic entity category?

Reply:
Not sure which one you are referring too,  I have completely migrated to a entity_description setup, so please have a look at that and let me, it includes categories for two sensors as well.